### PR TITLE
Add [Http|Grpc]LifecycleObserver examples

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
HttpLifecycleObserver and GrpcLifeCycleObserver interfaces were recently
introduced along with logger implementations. We don't have examples
which demonstrate their usage.

Modifications:
- Add grpc and http observer examples focused on the usage of
  HttpLifecycleObserver and GrpcLifeCycleObserver.

Result:
Example exist that demonstrate the new features.

### CODE REVIEW

File: servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/LifecycleObserverClient.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import io.grpc.examples.observer.Greeter.BlockingGreeterClient;
+import io.grpc.examples.observer.Greeter.ClientFactory;
+import io.grpc.examples.observer.HelloRequest;
+
+import static io.servicetalk.logging.api.LogLevel.TRACE;
+
+/**
+ * An example client that shows {@link GrpcLifecycleObserver} usage.
+ */
+public final class LifecycleObserverClient {
```
Comment (Scottmitch) [inline servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/LifecycleObserverClient.java:32]: I first just enhanced the `debugging` example and added these filters, however the logging filters aren't necessary specific to debugging so I created new/independent examples. I figured we could also leverage these examples for future observers (metrics/etc.)

File: servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/LifecycleObserverClient.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public static void main(String... args) throws Exception {
+        try (BlockingGreeterClient client = GrpcClients.forAddress("localhost", 8080)
+                // Append this filter first for most cases to maximize visibility!
```
Comment (idelpivnitskiy) [inline servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/LifecycleObserverClient.java:35]: Consider suggesting users to look at the filter javadoc, it clarifies visibility at different positions

File: servicetalk-examples/http/observer/build.gradle [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+apply plugin: "java"
+apply from: "../../gradle/idea.gradle"
+
+dependencies {
+  implementation project(":servicetalk-annotations")
+  implementation project(":servicetalk-http-netty")
+  implementation project(":servicetalk-http-utils") //
```
Comment (idelpivnitskiy) [inline servicetalk-examples/http/observer/build.gradle:23]: Unnecessary `//`

File: servicetalk-examples/http/observer/src/main/java/io/servicetalk/examples/http/observer/LifecycleObserverClient.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * An example client that shows {@link HttpLifecycleObserver} usage.
+ */
+public final class LifecycleObserverClient {
+    public static void main(String[] args) throws Exception {
+        // Note: this example demonstrates only blocking-aggregated programming paradigm, for asynchronous and
+        // streaming API see helloworld examples.
+        try (BlockingHttpClient client = HttpClients.forSingleAddress("localhost", 8080)
+                // Append this filter first for most cases to maximize visibility!
```
Comment (idelpivnitskiy) [inline servicetalk-examples/http/observer/src/main/java/io/servicetalk/examples/http/observer/LifecycleObserverClient.java:34]: Consider suggesting users to look at the filter javadoc, it clarifies visibility at different positions

File: servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/package-info.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
```
Comment (idelpivnitskiy) [inline servicetalk-examples/grpc/observer/src/main/java/io/servicetalk/examples/grpc/observer/package-info.java:2]: 2021

File: servicetalk-examples/http/observer/build.gradle [hunk 6]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
```
Comment (idelpivnitskiy) [inline servicetalk-examples/http/observer/build.gradle:2]: 2021

**Review by idelpivnitskiy — APPROVED** [review]

**Comment by Scottmitch** [comment]
0.41 (e33210a34)

**Pull request merged**